scripts/prepare_sector_network.py

```diff
@@ -157,6 +157,113 @@ def add_generation(
+                * (1 - costs.at[coal_CC_names[cc_tech], "capture_rate"]),
+                efficiency3=costs.at["coal", "CO2 intensity"]
+                * costs.at[coal_CC_names[cc_tech], "capture_rate"],
+                lifetime=costs.at[coal_CC_names[cc_tech], "lifetime"],
+            )
+
+    # add coal IGCC if enabled
+    if options["coal"].get("coal_IGCC", False):
+        n.madd(
+            "Link",
+            spatial.nodes + " " + "coal IGCC",
+            bus0=spatial.coal.nodes,
+            bus1=spatial.nodes,
+            bus2="co2 atmosphere",
+            marginal_cost=costs.at["Coal-IGCC", "efficiency"]
+            * costs.at["Coal-IGCC", "VOM"],  # NB: VOM is per MWel
+            # NB: fixed cost is per MWel
+            capital_cost=costs.at["Coal-IGCC", "efficiency"]
+            * costs.at["Coal-IGCC", "fixed"],
+            p_nom_extendable=True,
+            carrier="coal",
+            efficiency=costs.at["Coal-IGCC", "efficiency"],
+            efficiency2=costs.at["coal", "CO2 intensity"],
+            lifetime=costs.at["Coal-IGCC", "lifetime"],
+        )
+
+    # add natural gas CC technologies
+    if options["gas"].get("CC", False):
+        gas_CC_techs = options["gas"].get("gas_CC_techs", list())
+
+        gas_CC_names = {
+            "gas NGCC CC-95": "NG 2-on-1 Combined Cycle (F-Frame) 95% CCS",
+            "gas NGCC CC-97": "NG 2-on-1 Combined Cycle (F-Frame) 97% CCS",
+        }
+
+        for cc_tech in gas_CC_techs:
+            n.madd(
+                "Link",
+                spatial.nodes + " " + cc_tech,
+                bus0=spatial.gas.nodes,
+                bus1=spatial.nodes,
+                bus2="co2 atmosphere",
+                bus3=spatial.co2.nodes,
+                marginal_cost=costs.at[gas_CC_names[cc_tech], "efficiency"]
+                * costs.at[gas_CC_names[cc_tech], "VOM"],  # NB: VOM is per MWel
+                # NB: fixed cost is per MWel
+                capital_cost=costs.at[gas_CC_names[cc_tech], "efficiency"]
+                * costs.at[gas_CC_names[cc_tech], "fixed"],
+                p_nom_extendable=True,
+                carrier="gas",
+                efficiency=costs.at[gas_CC_names[cc_tech], "efficiency"],
+                efficiency2=costs.at["gas", "CO2 intensity"]
+                * (1 - costs.at[gas_CC_names[cc_tech], "capture_rate"]),
+                efficiency3=costs.at["gas", "CO2 intensity"]
+                * costs.at[gas_CC_names[cc_tech], "capture_rate"],
+                lifetime=costs.at[gas_CC_names[cc_tech], "lifetime"],
+            )
+
+    # add Natural Gas Combined Cycle plants if enabled
+    if options["gas"].get("gas_NGCC", False):
+        n.madd(
+            "Link",
+            spatial.nodes + " " + "gas NGCC",
+            bus0=spatial.gas.nodes,
+            bus1=spatial.nodes,
+            bus2="co2 atmosphere",
+            marginal_cost=costs.at["NG 2-on-1 Combined Cycle (F-Frame)", "efficiency"]

```

Comment:
Yes, I agree.

Reply:
Ahhh, I see. My major concern is linked with having the technologies parameters kind-of hard-coded in the script. That is also connected with what we have discussed for `gas NGCC CC` technologies above. My feeling is that it would be great to find a way to place for all the "advanced" technologies the names and "nice names" in the config for facilitate reviewing and modifying them. Does it make sense for you?

Reply:
Thanks, @ekatef, for suggestion. I have implemented you suggestions of removing dictionary inside `prepare_sector_script`. What I have changed is for list of `coal_CC_techs` and `gas_CC_techs`, I have used full CC technologies as you have suggested. However, technologies without carbon capture such as `gas_NGCC` and `coal_IGCC` I have left with shorter name, because they are keys of config not values. But to make it clear I have added explicit name as the comment both in the config and `prepare_sector_networks.py`. I think it should prevent misunderstanding.

Reply:
Sounds like a good solution for now, I think 🙂 Agree that it doesn't make to overcomplicate the implementation for `gas_NGCC` and `coal_IGCC` since there are single versions of the technologies